drivers/spi/spi_ll_stm32.c

```diff
@@ -171,7 +171,7 @@ static void spi_stm32_complete(struct spi_stm32_data *data, SPI_TypeDef *spi,
 
 	spi_context_cs_control(&data->ctx, false);
 
-#if defined(CONFIG_SOC_SERIES_STM32L4X) || defined(CONFIG_SOC_SERIES_STM32F3X)
+#if defined(CONFIG_SOC_SERIES_STM32L4X) || defined(CONFIG_SOC_SERIES_STM32F0X) || defined(CONFIG_SOC_SERIES_STM32F3X)
 	/* Flush RX buffer */
```

Comment:
80 characters limit

Reply:
Not related to only this PR:
I've come across this inmultiple stm32 driver files. Since it's basicly the same groups everywhere,
 that share the same driver behaviour (L4, F0, F3, F7 vs F1, F2, F4, L1) can't we add something further up the tree like CONFIG_STM32_SOC_GROUP_1 and CONFIG_STM32_SOC_GROUP_2. 
Selecting e.g. CONFIG_SOC_SERIES_STM32F4X would also select GROUP_1, selecting CONFIG_SOC_SERIES_STM32F0X would also select GROUP_2.

We'd avoid having to make these changes to driver files, when the driver itself is fine, but the defines are not aware of this SOC series now also supporting it.

Reply:
Yes, but it should be something like CONFIG_STM32_SPI_HAS_FIFO for F0/F3/L4/F7

Reply:
@DWagenk @erwango I added it, what do you think ?

Reply:
definitely improves readability in this driver. 
My approach was more general, because GROUP_2 (F0/F3/L4/F7) have spi-fifos, use i2c driver v2 and probably theres more, that this group has in common.
So I was targeting at something more general. Anybody with deep knowledge of the stm32 families here to answer, if there's enough similarities inside those groups, to introduce this define?

Reply:
I'm not sure we can draw precisely general groups as you propose. 
We only see a small subset of IPs in Zephyr for now, looking into whole range of IP, discrepancies will appear and there are more series coming and some new IP versions that will show up. Even if working for the IPs present today in Zephyr, this will soon be quite difficult to maintain.
But definitely we could group series into variation of one IP. Like it was done for I2C. Besides, this matches Zephyr hierarchy.